js/piwik.js

```diff
@@ -3678,6 +3698,34 @@ if (typeof Piwik !== 'object') {
                     }
                 }
 
+                var customDimensionIdsAlreadyHandled = [];
+                if (customData) {
```

Comment:
One can use `var customData = {dimension1:'value', dimension2:'value', ...}` with pretty much any action. We won't document it for now though. Only on request. 

`setCusomDimension(id, value)` these values will be used across all tracking requests no matter what.

If one wants to send a custom dimension only for eg one pageview one can do `trackPageview('title', {dimension1: 'value'})`.  

@mattab I think we should actually document `customData` as sending dimensions along with all requests will most likely result in unexpected results.